src/Hl7.Fhir.Base/Serialization/FhirJsonPocoDeserializer.cs

```diff
@@ -675,6 +675,7 @@ FhirJsonPocoDeserializerState state
                                 !Settings.DisableBase64Decoding ? readBase64(ref reader) : new(reader.GetString(), null),
                 JsonTokenType.String when requiredType == typeof(DateTimeOffset) => readDateTimeOffset(ref reader),
                 JsonTokenType.String when requiredType.IsEnum => new(reader.GetString(), null),
+                JsonTokenType.String when requiredType == typeof(long) => readLong(ref reader),
```

Comment:
When writing the parser I assumed longs are also serialized as json numbers, not strings - is that incorrect ? then we must probably adapt tryGetMatchingNumber - since long handling was done there, and we should trigger an error if the model says something is a long, but the json serialization uses a non-string.

Reply:
You'd say based on http://hl7.org/fhir/json.html#primitive that longs are indeed strings. I wonder if this is intentional. Anyway, you probably found an example that does this (thus triggering this change ;-)).

Reply:
See my general comment in this PR.